datalad/distributed/drop.py

```diff
@@ -0,0 +1,792 @@
+    jobs_opt,
+    recursion_flag,
+    recursion_limit,
+)
+from datalad.interface.results import (
+    annexjson2result,
+    results_from_annex_noinfo,
+    success_status_map,
+)
+from datalad.interface.utils import eval_results
+from datalad.runner.exception import CommandError
+from datalad.support.annexrepo import AnnexRepo
+from datalad.support.constraints import (
+    EnsureChoice,
+    EnsureNone,
+    EnsureStr,
+)
+from datalad.support.exceptions import CapturedException
+from datalad.support.param import Parameter
+from datalad.utils import (
+    ensure_list,
+    rmtree,
+)
+
+lgr = logging.getLogger('datalad.distributed.drop')
+
+
+@build_doc
+class Drop(Interface):
+    """Drop content of individual files or entire (sub)datasets
+
+    This command is the antagonist of 'get'. It can undo the retrieval of file
+    content, and the installation of subdatasets.
+
+    Dropping is a safe-by-default operation. Before dropping any information,
+    the command confirms the continued availability of file-content (see e.g.,
+    configuration 'annex.numcopies'), and the state of all dataset branches
+    from at least one known dataset sibling. Moreover, prior removal of an
+    entire dataset annex, that it is confirmed that it is no longer marked
+    as existing in the network of dataset siblings.
+
+    Importantly, all checks regarding version history availability and local
+    annex availability are performed using the current state of remote
+    siblings as known to the local dataset. This is done for performance
+    reasons and for resilience in case of absent network connectivity. To
+    ensure decision making based on up-to-date information, it is advised to
+    execute a dataset update before dropping dataset components.
+    """
+    _examples_ = [
+        {'text': "Drop single file content",
+         'code_py': "drop('path/to/file')",
+         'code_cmd': "datalad drop <path/to/file>"},
+        {'text': "Drop all file content in the current dataset",
+         'code_py': "drop('.')",
+         'code_cmd': "datalad drop"},
+        {'text': "Drop all file content in a dataset and all its subdatasets",
+         'code_py': "drop(dataset='.', recursive=True)",
+         'code_cmd': "datalad drop -d <path/to/dataset> -r"},
+        {'text': "Disable check to ensure the configured minimum number of "
+                 "remote sources for dropped data",
+         'code_py': "drop(path='path/to/content', reckless='availability')",
+         'code_cmd': "datalad drop <path/to/content> --reckless availability"},
+        {'text': 'Drop (uninstall) an entire dataset '
+                 '(will fail with subdatasets present)',
+         'code_py': "drop(what='all')",
+         'code_cmd': "datalad drop --what all"},
+        {'text': 'Kill a dataset recklessly with any existing subdatasets too'
+                 '(this will be fast, but will disable any and all safety '
+                 'checks)',
```

Comment:
```suggestion
        {'text': 'Kill a dataset recklessly with any existing subdatasets too '
                 '(this will be fast, but will disable any and all safety '
                 'checks)',
```

There is a missing space:

```
Kill a dataset recklessly with any existing subdatasets too(this will
be fast, but will disable any and all safety checks)::
```